src/xenia/cpu/backend/x64/x64_seq_vector.cc

```diff
@@ -731,6 +731,25 @@ struct VECTOR_SHL_V128
   static void EmitInt8(X64Emitter& e, const EmitArgType& i) {
     // TODO(benvanik): native version (with shift magic).
     if (i.src2.is_constant) {
+      if (e.IsFeatureEnabled(kX64EmitGFNI)) {
+        const auto& shamt = i.src2.constant();
+        bool all_same = true;
+        for (size_t n = 0; n < 16 - n; ++n) {
+          if (shamt.u8[n] != shamt.u8[n + 1]) {
+            all_same = false;
+            break;
+          }
+        }
+        if (all_same) {
+          // Every count is the same, so we can use gf2p8affineqb.
+          const uint8_t shift_amount = shamt.u8[0];
+          const uint64_t shift_matrix =
+              0x0102040810204080 >> (shift_amount * 8);
```

Comment:
Missing `ull` (or `ULL`) on constants again. 🙂

Reply:
Or better `UINT64_C(0x0102040810204080)` since `ULL` is for `unsigned long long`, not specifically `uint64_t`.

Reply:
By the way, can the shift amount overflow, and is that handled according to the PowerPC specification?

Reply:
That's fair, I wasn't aware of `UINT64_C`.

Reply:
> By the way, can the shift amount overflow, and is that handled according to the PowerPC specification?

![chrome_Zubs3WCDOJ](https://user-images.githubusercontent.com/644247/150817849-a1b2e619-8a95-4835-b8ea-fbc79eb35ea7.png)

[AltiVec](https://www.nxp.com/docs/en/reference-manual/ALTIVECPEM.pdf) doc seems to say that it just masks the lower `log2(n)` bits. I'll do the masking and const-literal stuff in a new commit.